demisto_sdk/commands/validate/validators/PB_validators/PB101_is_playbook_has_unreachable_condition.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import annotations
+
+from typing import Iterable, List
+
+from demisto_sdk.commands.content_graph.objects.playbook import Playbook
+from demisto_sdk.commands.validate.validators.base_validator import (
+    BaseValidator,
+    ValidationResult,
+)
+
+ContentTypes = Playbook
+
+
+class IsAskConditionHasUnreachableConditionValidator(BaseValidator[ContentTypes]):
+    error_code = "PB101"
+    description = "Checks whether an ask conditional has task with unreachable next task condition."
+    rationale = "Checks whether an ask conditional has task with unreachable next task condition."
+    error_message = (
+        "Playbook conditional task with id:{task_id} has task with unreachable "
+        "next task condition '{next_task_branch}'. Please remove this task or add "
+        "this condition to condition task with id:{task_id}."
+    )
+    related_field = "tasks"
+    is_auto_fixable = False
+
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        """Checks whether a builtin conditional task branches are handled properly
+        Args:
+            - content_items (Iterable[ContentTypes]): The content items to check.
+        Return:
+            - List[ValidationResult]. List of ValidationResults objects.
+        """
+        results: List[ValidationResult] = []
+        for content_item in content_items:
+            invalid_tasks = self.invalid_tasks(content_item)
```

Comment:
There are allot of nested for loops. is_valid has 2 from which one is calling invalid_tasks which have 3 more nested loops.
the result append is not dependent on invalid_tasks.
append all invalid_tasks in a single for loop.
Then append all items from invalid_tasks to resutls not in a nested for loop.

Reply:
Seems like there is a dependency down the road which prevents breaking the nested for loop